mlir/tools/rocmlir-gen/rocmlir-gen.cpp

```diff
@@ -3373,6 +3378,51 @@ static ModuleOp generateKernel(MLIRContext *context, GenParams &genParams,
   return module;
 }
 
+static ModuleOp populateCloneHarnessLogic(ModuleOp &module) {
+  if (arch.getValue().empty()) {
```

Comment:
In a MIGraphX context, the arch may be sitting on the function inside the module. Do you want to go looking for it, or will we be expected to copy-paste if need be?

Reply:
If the function already has the arch, I think I can go find it, probably in the next PR.